contracts/Core/AssetManager.sol

```diff
@@ -82,37 +81,28 @@ contract AssetManager is AssetStorage, StateManager, AssetManagerParams, IAssetM
         uint32 epoch = _getEpoch(epochLength);
         if (assetStatus) {
             if (!collections[id].active) {
-                activeCollections.push(id);
-                collections[id].assetIndex = uint16(activeCollections.length);
+                // slither-disable-next-line incorrect-equality
+                if (updateRegistry == epoch) {
+                    // slither-disable-next-line reentrancy-events,reentrancy-no-eth,reentrancy-benign
+                    delegator.updateRegistry(numCollections);
+                }
+                numActiveCollections = numActiveCollections + 1;
                 collections[id].active = assetStatus;
+                updateRegistry = epoch + 1;
```

Comment:
hey what does updateRegistry mean ?

Reply:
Epoch in which the registry needs to be updated. I think i should rename to ``updateRegistryEpoch``